### ISSUE TITLE
Support for using some crates with SP1

### ISSUE THREAD

**Issue by manav2401** [alloy-rs/alloy#1112]
### Component

other

### Describe the feature you would like

I want to use `alloy-primitives` and `alloy-rlp` as a dependency in SP1 but it doesn't compile. It'll be great if [this](https://github.com/alloy-rs/alloy/blob/fa01c2ab9174abcd9521624ef341fcfea3d08356/scripts/check_no_std.sh#L12) command worked for these 2 crates.

Thanks!

**Comment by DaniPopes** [comment]
Can you please elaborate on "it doesn't compile"? All of the crates in that script depends on either alloy-primitives, alloy-rlp, or both, so by default these also compile using that command.

**Comment by manav2401** [comment]
> All of the crates in that script depend on either alloy-primitives, alloy-rlp, or both, so by default these also compile using that command.

Ahh didn't know that, thanks!

> Can you please elaborate on "it doesn't compile"?

Sure. Basically I added `reth-primitives`, `alloy-primitives` and `alloy-rlp` as a dependency in my SP1 project. While the root rust program compiles well, SP1's zkVM can't really compile the programme for generating proofs. I had a discussion with their team and they mentioned that they maintain a custom fork for such compatibility changes. I thought it will be great to have those custom changes (which I am not fully aware of) in the main repo itself. They mentioned that they'll create these PRs and they'll also update the CI such that any changes which affects their lifecycle gets caught. 

TL;DR: This might be something which their team will take a look into. Thanks!
